src/main/java/staffconnect/model/person/Faculty.java

```diff
@@ -0,0 +1,113 @@
+package staffconnect.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static staffconnect.commons.util.AppUtil.checkArgument;
+
+import java.util.Arrays;
+
+/**
+ * Represents a Person's faculty in the staff book.
+ * Guarantees: immutable; is valid as declared in
+ * {@link #isValidFaculty(String)}
+ */
+public class Faculty {
+    public static final String MESSAGE_CONSTRAINTS =
+            "The content should be a String representing a real faculty of NUS";
+
+    /**
+     * For this version, a valid faculty value should match exactly the full name of the facult,
+     * or the value is invalid
+     */
+    public enum FacultyEnum {
+        ARTS_AND_SOCIAL_SCIENCES("Arts and Social Sciences"),
+        BUSINESS("Business"),
+        COMPUTING("Computing"),
+        CONTINUING_AND_LIFELONG_EDUCATION("Continuing and Lifelong Education"),
+        DENTISTRY("Dentistry"),
+        DESIGN_AND_ENVIRONMENT("Design and Environment"),
+        DUKE_NUS_MEDICAL_SCHOOL("Duke-NUS Medical School"),
+        ENGINEERING("Engineering"),
+        INTEGRATIVE_SCIENCES_AND_ENGINEERING("Integrative Sciences and Engineering"),
+        LAW("Law"),
+        MEDICINE("Medicine"),
+        MUSIC("Music"),
+        PUBLIC_HEALTH("Public Health"),
+        PUBLIC_POLICY("Public Policy"),
+        SCIENCE("Science"),
+        UNIVERSITY_SCHOLARS_PROGRAMME("University Scholars Programme"),
+        YALE_NUS_COLLEGE("Yale-NUS College");
+
+        private final String facultyName;
+
+        FacultyEnum(String facultyName) {
+            this.facultyName = facultyName;
+        }
+
+        /**
+         * Links enum member to its name.
+         *
+         * @return name of the faculty
+         */
+        public String getFacultyName() {
+            return facultyName;
+        }
+    }
+    public final String value;
+
+    /**
+     * Constructs a {@code Faculty}.
+     *
+     * @param faculty A valid faculty.
+     */
+    public Faculty(String faculty) {
+        requireNonNull(faculty);
+        checkArgument(isValidFaculty(faculty), MESSAGE_CONSTRAINTS);
+        value = fromString(faculty).getFacultyName(); // can be extended
```

Comment:
So far so good, though the logic is string -> enum->string, it makes having the enum class a bit redundant. May I suggest storing the enum instead

Reply:
I did consider that, but I wish to keep conformity as other attributes which are also strings. The fromString() function can be extended so that maybe syncronyms can be allowed. don't have to type in the full name of the faculty. Adding enum can simplify the process.